crates/autopilot/src/database/competition.rs

```diff
@@ -0,0 +1,136 @@
+use {
+    anyhow::Context,
+    database::{
+        auction::AuctionId,
+        auction_participants::Participant,
+        auction_prices::AuctionPrice,
+        byte_array::ByteArray,
+        settlement_scores::Score,
+    },
+    model::order::OrderUid,
+    number_conversions::u256_to_big_decimal,
+    primitive_types::{H160, U256},
+    std::collections::{BTreeMap, HashSet},
+};
+
+#[derive(Clone, Debug)]
+pub enum ExecutedFee {
+    Solver(U256),
+    /// Optionable because, for partially fillable limit orders, surplus fee is
+    /// unknown until the transaction is mined.
+    Surplus(Option<U256>),
+}
+
+#[derive(Clone, Debug)]
+pub struct OrderExecution {
+    pub order_id: OrderUid,
+    pub executed_fee: ExecutedFee,
+}
+
+#[derive(Clone, Default)]
+pub struct Competition {
+    pub auction_id: AuctionId,
+    pub winner: H160,
+    pub winning_score: U256,
+    pub reference_score: U256,
+    /// Addresses to which the CIP20 participation rewards will be payed out.
+    /// Usually the same as the solver addresses.
+    pub participants: HashSet<H160>,
```

Comment:
I think participants needs to be distinct from the payout address. For example a solver might request to change the address in the middle of the week. If in that case we had a participation reward computation logic that requires at least `n` submissions to qualify for a payout the solver would receive less than they reasonably should.

Reply:
Same as here: https://github.com/cowprotocol/services/pull/1425#discussion_r1198807045